gsoc/models.py

```diff
@@ -0,0 +1,222 @@
+    slots_can_be_requested = models.BooleanField()
+    org_accept_reject_emails_can_be_sent = models.BooleanField()
+    org_payment_details_visible = models.BooleanField()
+    students_can_submit_proposals = models.BooleanField()
+    student_withdrawals_trigger_email = models.BooleanField()
+    enrollment_forms_can_be_re_submitted = models.BooleanField()
+    tax_forms_can_be_reviewed = models.BooleanField()
+    first_eval_emails_can_be_sent = models.BooleanField()
+    second_review_status_visible = models.BooleanField()
+    project_decisions_can_be_finalized = models.BooleanField()
+    student_home_address_is_editable = models.BooleanField()
+    program = models.ForeignKey(Program, on_delete=models.CASCADE)
+
+    def __str__(self):
+        return str(self.program)
+
+
+class ProgramStatistic(models.Model):
+    number_of_lines_of_code = models.IntegerField()
+    homepage_start_button = models.CharField(max_length=100)
+    homepage_intro_paragraph = models.TextField()
+    number_of_mentors = models.IntegerField()
+    number_of_student_and_mentor_countries = models.IntegerField()
+    number_of_years = models.IntegerField()
+    homepage_orgs_paragraph = models.TextField()
+    homepage_students_paragraph = models.TextField()
+    number_of_organizations = models.IntegerField()
+    number_of_student_countries = models.IntegerField()
+    program = models.ForeignKey(Program, on_delete=models.CASCADE)
+
+    def __str__(self):
+        return self.homepage_start_button
+
+
+class Organization(models.Model):
+    name = models.CharField(max_length=100, primary_key=True)
+    logo = models.ImageField(upload_to='', blank=True)
+    url = models.URLField(blank=True)
+    summary = models.TextField(max_length=300, default='')
+    technologies = models.ManyToManyField('Technology')
+    topics = models.ManyToManyField('Topic')
+
+    def __str__(self):
+        return self.name
+
+
+class Technology(models.Model):
+    name = models.CharField(max_length=100)
+
+    def __str__(self):
+        return self.name
+
+
+class Topic(models.Model):
+    name = models.CharField(max_length=100)
+
+    def __str__(self):
+        return self.name
+
+
+class Project(models.Model):
+    name = models.CharField(max_length=100, primary_key=True)
+    summary = models.TextField(max_length=300, default='')
+    org = models.ForeignKey(Organization, on_delete=models.CASCADE)
+    student = models.OneToOneField('Student')
+    mentor = models.ForeignKey('Mentor', on_delete=models.CASCADE)
+    program = models.ForeignKey(Program, on_delete=models.CASCADE)
+
+    def __str__(self):
+        return self.name
+
+
+class Student(models.Model):
+    name = models.CharField(max_length=100)
+
+    def __str__(self):
+        return self.name
+
+
+class Mentor(models.Model):
+    name = models.CharField(max_length=100)
+
+    def __str__(self):
+        return self.name
+
+
+# for 2018 events
+
+
+class UpcomingEvent(models.Model):
```

Comment:
these should be GsocEvent/ GsocOccurrence - they are the same events that are found in the `Program` json